searx/engines/prowlarr.py

```diff
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+
+ Prowlarr search (Files)
+
+"""
+
+from json import loads
+from urllib.parse import urlencode
+from searx.exceptions import SearxEngineAPIException
+
+categories = ''
+
+paging = False
+api_key = ''
+indexer_ids = ''
+search_type = 'search'
+search_categories = ''
+base_url = ''
+
+
+def request(query, params):
+    if not base_url:
+        raise SearxEngineAPIException('missing prowlarr base url')
+
+    if not api_key:
+        raise SearxEngineAPIException('missing prowlarr API key')
+
+    query_args = {
+        'query': query,
+        'apikey': api_key,
+        'type': search_type
+    }
+
+    if indexer_ids:
+        query_args['indexerIds'] = indexer_ids
+
+    if search_categories:
+        query_args['categories'] = search_categories
+
+    params['url'] = base_url + urlencode(query_args)
+
+    return params
+
+
+def response(resp):
+    results = []
+
+    json_data = loads(resp.text)
+
+    for result in json_data:
+
+        newResult = {
+            'title': result['title'],
+            'url': result['infoUrl'],
+            'template': 'torrent.html'
+        }
+
+        if 'files' in result:
+            newResult['files'] = result['files']
```

Comment:
Nit: please use snake_case in variable names.

Reply:
@kvch oops.. sorry.. this PR and #3119 are actually my first python things i've written..

i'm coming from java and php, where camelCase rules.. it's deeply ingrained into my muscle memory..